core/resharding.go

```diff
@@ -19,12 +18,16 @@ import (
 const (
 	// InitialSeed is the initial random seed, a magic number to answer everything, remove later
 	InitialSeed uint32 = 42
-	// GenesisEpoch is the number of the first genesis epoch.
+	// GenesisEpoch is the number of the genesis epoch.
 	GenesisEpoch = 0
+	// FirstEpoch is the number of the first epoch.
+	FirstEpoch = 1
 	// GenesisShardNum is the number of shard at genesis
-	GenesisShardNum = 3
+	GenesisShardNum = 4
 	// GenesisShardSize is the size of each shard at genesis
 	GenesisShardSize = 10
+	// CuckooRate is the percentage of nodes getting reshuffled in the second step of cuckoo resharding.
+	CuckooRate = 0.1
```

Comment:
If we use constant rate, how to make sure it's safe when number of new nodes is greater than 10% ?

Reply:
It will get stable after a few epochs. The thinking is that with previous kickout calculation, if a lot of nodes joins at once, it's a risk to disturb the network a lot.